### ISSUE TITLE
Incorrect fuel distributor warning after restart

### ISSUE THREAD

**Issue by colingeniet** [NikolaiVChr/flightgear-saab-ja-37-viggen#77]
Steps to reproduce: 
- crash the aircraft
- restart from airport (location->select airport)
- start engine
- ensure >43% fuel and gear down

The fuel distrib. warning triggers. Restarting a second time fixes the issue.

Looking at the code and properties, when the issue occurs `/fdm/jsbsim/systems/indicators/fuel-flow-primary` is 1, yet `/fdm/jsbsim/systems/fuel/pneumatic-low-pressure` is also 1, which makes no sense since the definition of `systems/indicators/fuel-flow-primary` is
```
<and>
    [...]
    <eq>
        <property>systems/fuel/pneumatic-low-pressure</property>
         <value>0</value>
    </eq>
</and>
```

**Comment by NikolaiVChr** [comment]
I can reproduce in FG 2019.1.1.

Very strange. I do not understand it.

**Comment by NikolaiVChr** [comment]
When I retract gear I still get 1, its like its stuck and not running.

**Comment by NikolaiVChr** [comment]
If in JSB file I reorder the included files like this then the error goes away:

```
  <propulsion file="Systems/jsb-propulsion-ja37"/>
  <flight_control file="Systems/jsb-controls"/>
  <aerodynamics file="ja37-aerodynamics.xml"/>
  <system file="jsb-hydraulic"/>
  <system file="jsb-electric"/>
  <system file="jsb-cannon_recoil"/>
  <system file="jsb-fuel"/>
  <external_reactions file="Systems/jsb-external-reactions"/>
  <ground_reactions file="Systems/jsb-ground-reactions"/>
  <system file="jsb-flight"/>
  <system file="jsb-indicators"/>
```

**Comment by NikolaiVChr** [comment]
I will report this on the devel list.

**Comment by colingeniet** [comment]
I found that replacing the `<fcs_function>` by a corresponding `<switch>` also works around the issue.

**Comment by zakalawe** [comment]
'Select airport' does a *relocate*, not a full-reset. So I think this maybe an actual JSBsim bug? Especially if re-ordering or changing the function to a switch, changes the observed behaviour?

Of course it could still be in FlightGear, I'm just less sure where+how if it's due to a relocate.

**Comment by NikolaiVChr** [comment]
Thank you, I will report on JSBSIM GitHub. :)

**Comment by bcoconni** [comment]
@zakalawe and @NikolaiVChr 

I don't see how you guys jump to the conclusion that it is a bug in JSBSim ?
1. I have no clue what a *relocate* is. This is FlightGear function, how can you infer from that there is a bug in JSBSim ?
2. The fact that the bug is sensitive to the order in which the functions are specified is by no means an evidence that the bug lies in JSBSim. It sounds just like you would pretend that a C++ compiler is buggy because the result of a program depends on the order in which the C++ code is written !!!

**Comment by NikolaiVChr** [comment]
Roger, I guess its in FG then.

I will port the aircraft over to FGAddon now. With the workaround applied.

I hope this somehow gets solved though, it might affect other systems and aircraft as well.

**Comment by bcoconni** [comment]
Sorry my initial answer was too harsh and I apologize for that.
> I guess its in FG then.

At this stage there is no evidence to support this statement: the fact that the result depends on the order in which the systems are specified does not prove there is a bug either in JSBSim or in FlightGear.

It rather looks like the property is overwritten by one of the systems you had to reorder to get a successful restart. I can only suggest you to start your investigations looking for such an occurrence.

**Comment by NikolaiVChr** [comment]
It looks like this is fixed on FG next. Yay!

At least I could not reproduce. And I have on purpose not applied the workaround so could keep track on this bug.

**Comment by colingeniet** [comment]
Closing this since the issue was in FG and is solved since FG 2019.2.
